### PR TITLE
Adds -run as an input to the analyzer

### LINKED ISSUE #995: [RFC #0105] Analyzer changes for run image extension (Dockerfiles phase 3)
Spec PR: https://github.com/buildpacks/spec/pull/347/files

For newer platform API (`0.12` and above), the `analyzer` accepts `-run` instead of `-stack`. `-run` is the location of a `run.toml` file with schema:

```
[[image]]
 image = "<image>"
 mirrors = ["<mirror1>", "<mirror2>"]
```

If no `-run-image` is provided, the analyzed run image should be the first image in `run.toml`.

### PR DESCRIPTION
**Pull request by natalieparellano**
Fixes #995

### CODE REVIEW

File: platform/analyze_inputs.go [hunk 1]
```diff
@@ -27,6 +27,12 @@ func DefaultAnalyzeInputs(platformAPI *api.Version) LifecycleInputs {
 }
 
 func defaultAnalyzeInputs() LifecycleInputs {
+	ai := defaultAnalyzeInputs08To011()
```
Comment (natalieparellano) [inline platform/analyze_inputs.go:30]:
```suggestion
	ai := defaultAnalyzeInputs09To011()
```

**Comment by natalieparellano** [comment]
@joe-kimmel-vmw this might intersect a bit with the changes in https://github.com/buildpacks/lifecycle/pull/994, but I'll hold off on trying to merge this one until that one is through

**Comment by natalieparellano** [comment]
This also needs to include `-run` as an input to the `creator`

**Comment by joe-kimmel-vmw** [comment]
> ... but I'll hold off on trying to merge this one ...
Maybe?  Let's check in again next week; today I feel like I'm just adding scope to that other PR :)

File: platform/analyze_inputs_test.go [hunk 2]
```diff
@@ -79,15 +139,6 @@ func testAnalyzeInputs(platformAPI string) func(t *testing.T, when spec.G, it sp
...
 			})
-
-			when("layers directory is provided", func() {
-				it("writes analyzed.toml at the layers directory", func() {
-					inputs.LayersDir = "some-layers-dir"
-					err := platform.ResolveInputs(platform.Analyze, &inputs, logger)
-					h.AssertNil(t, err)
-					h.AssertEq(t, inputs.AnalyzedPath, filepath.Join("some-layers-dir", "analyzed.toml"))
-				})
-			})
```
Comment (natalieparellano) [inline platform/analyze_inputs_test.go:90]: This test was repeated in this file

**Review by jabrown85 — APPROVED** [review]

**Comment by natalieparellano** [comment]
@joe-kimmel-vmw I checked the merge conflict - and thankfully it's not as bad as I thought, just some log messages that were updated in analyzer.go. I'll go ahead and merge this one so that I can pick up other issues in the "extensions" track

**Pull request merged**